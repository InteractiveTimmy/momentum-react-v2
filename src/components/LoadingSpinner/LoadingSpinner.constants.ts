const CLASS_PREFIX = 'md-loading-spinner';

const DEFAULTS = {};

const STYLE = {
  wrapper: `${CLASS_PREFIX}-wrapper`,
};

export { CLASS_PREFIX, DEFAULTS, STYLE };
