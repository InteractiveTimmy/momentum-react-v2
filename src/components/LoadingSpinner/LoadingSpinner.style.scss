.md-loading-spinner-wrapper {
  position: relative;
  fill: var(--spinner-loading-track);

  @keyframes rotate {
    from {
      transform: rotate(0deg);
    }

    from {
      transform: rotate(360deg);
    }
  }

  .md-loading-spinner-arch {
    position: absolute;
    top: 0;
    animation-name: rotate;
    animation-duration: 0.75s;
    animation-iteration-count: infinite;
    animation-timing-function: linear;

    svg {
      fill: var(--spinner-loading-cursor);
    }
  }
}
