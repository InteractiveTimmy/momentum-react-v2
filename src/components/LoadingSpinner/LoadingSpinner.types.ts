import { CSSProperties, ReactNode } from 'react';

export interface Props {
  /**
   * Child components of this LoadingSpinner.
   */
  children?: ReactNode;

  /**
   * Custom class for overriding this component's CSS.
   */
  className?: string;

  /**
   * Custom id for overriding this component's CSS.
   */
  id?: string;

  /**
   * Custom style for overriding this component's CSS.
   */
  style?: CSSProperties;
}
