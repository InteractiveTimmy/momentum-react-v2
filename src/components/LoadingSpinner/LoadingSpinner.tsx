import React, { FC } from 'react';
import classnames from 'classnames';

import { STYLE } from './LoadingSpinner.constants';
import { Props } from './LoadingSpinner.types';
import './LoadingSpinner.style.scss';
import Icon from '../Icon';

// TODO: Update JSDOC for this component.
/**
 * The LoadingSpinner component.
 */
const LoadingSpinner: FC<Props> = (props: Props) => {
  const { className, id, style } = props;

  // TODO: Implementation goes here.
  return (
    <div className={classnames(className, STYLE.wrapper)} id={id} style={style}>
      <Icon scale={24} name="spinner-filled" weight="regular" fillColor="red" strokeColor="none" />
      <Icon scale={24} name="spinner" weight="regular" fillColor="red" strokeColor="none" />
    </div>
  );
};

export default LoadingSpinner;
