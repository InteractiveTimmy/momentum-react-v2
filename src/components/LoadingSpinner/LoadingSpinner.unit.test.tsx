import React from 'react';
import { mount } from 'enzyme';

import LoadingSpinner, { LOADING_SPINNER_CONSTANTS as CONSTANTS } from './';

describe('<LoadingSpinner />', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      expect.assertions(1);

      const container = mount(<LoadingSpinner />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with className', () => {
      expect.assertions(1);

      const className = 'example-class';

      const container = mount(<LoadingSpinner className={className} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with id', () => {
      expect.assertions(1);

      const id = 'example-id';

      const container = mount(<LoadingSpinner id={id} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with style', () => {
      expect.assertions(1);

      const style = { color: 'pink' };

      const container = mount(<LoadingSpinner style={style} />);

      expect(container).toMatchSnapshot();
    });

    /* ...additional snapshot tests... */
  });

  describe('attributes', () => {
    it('should have its wrapper class', () => {
      expect.assertions(1);

      const element = mount(<LoadingSpinner />)
        .find(LoadingSpinner)
        .getDOMNode();

      expect(element.classList.contains(CONSTANTS.STYLE.wrapper)).toBe(true);
    });

    it('should have provided class when className is provided', () => {
      expect.assertions(1);

      const className = 'example-class';

      const element = mount(<LoadingSpinner className={className} />)
        .find(LoadingSpinner)
        .getDOMNode();

      expect(element.classList.contains(className)).toBe(true);
    });

    it('should have provided id when id is provided', () => {
      expect.assertions(1);

      const id = 'example-id';

      const element = mount(<LoadingSpinner id={id} />)
        .find(LoadingSpinner)
        .getDOMNode();

      expect(element.id).toBe(id);
    });

    it('should have provided style when style is provided', () => {
      expect.assertions(1);

      const style = { color: 'pink' };
      const styleString = 'color: pink;';

      const element = mount(<LoadingSpinner style={style} />)
        .find(LoadingSpinner)
        .getDOMNode();

      expect(element.getAttribute('style')).toBe(styleString);
    });

    /* ...additional attribute tests... */
  });

  describe('actions', () => {
    /* ...action tests... */
  });
});
