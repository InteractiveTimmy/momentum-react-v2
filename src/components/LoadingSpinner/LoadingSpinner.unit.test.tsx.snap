// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingSpinner /> snapshot should match snapshot 1`] = `
<LoadingSpinner>
  <div
    className="md-loading-spinner-wrapper"
  />
</LoadingSpinner>
`;

exports[`<LoadingSpinner /> snapshot should match snapshot with className 1`] = `
<LoadingSpinner
  className="example-class"
>
  <div
    className="example-class md-loading-spinner-wrapper"
  />
</LoadingSpinner>
`;

exports[`<LoadingSpinner /> snapshot should match snapshot with id 1`] = `
<LoadingSpinner
  id="example-id"
>
  <div
    className="md-loading-spinner-wrapper"
    id="example-id"
  />
</LoadingSpinner>
`;

exports[`<LoadingSpinner /> snapshot should match snapshot with style 1`] = `
<LoadingSpinner
  style={
    Object {
      "color": "pink",
    }
  }
>
  <div
    className="md-loading-spinner-wrapper"
    style={
      Object {
        "color": "pink",
      }
    }
  />
</LoadingSpinner>
`;
